Lib/ensurepip/_structs.py

```diff
@@ -0,0 +1,251 @@
+        """Verify the content's SHA-256 hash against recorded value."""
+        return self.wheel_sha256 == _compute_sha256(wheel_content).hexdigest()
+
+    def __repr__(self) -> str:
+        """Render remote distribution package instance for humans."""
+        return (
+            f'{self.project_name !s} == {self.project_version !s}'
+            f' @ {self.verifiable_wheel_file_url !s}'
+        )
+
+
+@dataclass(frozen=True)
+class BundledDistributionPackage:
+    """Structure representing a wheel under ``ensurepip/_bundled/``."""
+
+    project_name: str
+    """PyPI project name."""
+
+    project_version: str
+    """PyPI project version."""
+
+    wheel_name: str
+    """Wheel package file name."""
+
+    wheel_path: _resources_abc.Traversable
+    """Wheel package file path."""
+
+    @classmethod
+    def from_project_spec(
+            cls,
+            project_name: str,
+            project_version: str,
+            /,
+    ) -> BundledDistributionPackage:
+        """Create a replacement package from name and version spec."""
+        wheel_name = _convert_project_spec_to_wheel_name(
+            project_name, project_version,
+        )
+        wheel_path = BUNDLED_WHEELS_PATH / wheel_name
+        return cls(project_name, project_version, wheel_name, wheel_path)
+
+    @classmethod
+    def from_remote_dist(
+            cls,
+            remote_distribution_pkg: _RemoteDistributionPackage,
+            /,
+    ) -> BundledDistributionPackage:
+        """Create a replacement package from its remote counterpart."""
+        return cls.from_project_spec(
+            remote_distribution_pkg.project_name,
+            remote_distribution_pkg.project_version,
+        )
+
+    def as_pathlib_ctx(self) -> AbstractContextManager[Path]:
+        """Make a context manager exposing a :mod:`pathlib` instance."""
+        return _traversable_to_pathlib_ctx(self.wheel_path)
+
+
+@dataclass(frozen=True)
+class ReplacementDistributionPackage:
+    """Structure representing a wheel under ``WHEEL_PKG_DIR``."""
+
+    project_name: str
+    """PyPI project name."""
+
+    project_version: str
+    """PyPI project version."""
+
+    wheel_name: str
+    """Wheel package file name."""
+
+    wheel_path: Path
+    """Wheel package file path."""
+
+    @classmethod
+    def from_distribution_package_name(
+            cls,
+            dist_pkg_name: str,
+            /,
+    ) -> ReplacementDistributionPackage:
+        """Look up a replacement package from name.
+
+        :raises LookupError: If ``WHEEL_PKG_DIR`` is not set or the package
+                             wheel is nowhere to be found.
+
+        """
+        wheel_file_name_prefix = f'{dist_pkg_name !s}-'
+
+        wheel_pkg_dir = _get_wheel_pkg_dir_from_sysconfig()
+        dist_matching_wheels = wheel_pkg_dir.glob(
+            f'{wheel_file_name_prefix !s}*.whl',
+        )
+
+        try:
+            first_matching_dist_wheel = sorted(dist_matching_wheels)[0]
```

Comment:
Self note: the original code actually goes for the last matching:
```suggestion
            last_matching_dist_wheel = sorted(dist_matching_wheels)[-1]
```